contracts/dtoken/src/common.rs

```diff
@@ -212,6 +212,10 @@ impl Contract {
     pub fn increase_contract_balance(&mut self, amount: WBalance) -> Balance {
         self.set_contract_balance(self.get_contract_balance() + Balance::from(amount))
     }
+
+    pub fn decrease_contract_balance(&mut self, amount: WBalance) -> Balance {
```

Comment:
There may be a negative value here?

Reply:
we've solved this issue by having Reserve action, which aims not to drain the contract's balance. 

yet this is a fair comment, I will add that assert there